docs/core/deploying/single-file.md

```diff
@@ -11,9 +11,26 @@ Bundling all application-dependent files into a single binary provides an applic
+## Impact of using SingleFile feature
 
-Single-file doesn't bundle native libraries by default. On Linux, we prelink the runtime into the bundle and only application native libraries are deployed to the same directory as the single-file app. On Windows, we prelink only the hosting code and both the runtime and application native libraries are deployed to the same directory as the single-file app. This is to ensure a good debugging experience, which requires native files to be excluded from the single file. There is an option to set a flag, `IncludeNativeLibrariesForSelfExtract`, to include native libraries in the single file bundle, but these files will be extracted to a temporary directory in the client machine when the single file application is run.
+There are some caveats that you need to be aware for single-file use, first of which is the use of path information to locate a file relative to the location of your application. The <xref:System.Reflection.Assembly.Location?displayProperty=nameWithType> API will return an empty string, which is the default behavior for assemblies loaded from memory. The compiler will give a warning for this API during build time to alert the developer to the specific behavior. If the path to the application directory is needed, the <xref:System.AppContext.BaseDirectory?displayProperty=nameWithType> API will return the directory where the AppHost (the single-file bundle itself) resides. The table at the bottom of this section has runtime library API behavior in single-file use.
+
+Single-file doesn't bundle native libraries by default. On Linux, we prelink the runtime into the bundle and only application native libraries are deployed to the same directory as the single-file app. On Windows, we prelink only the hosting code and both the runtime and application native libraries are deployed to the same directory as the single-file app. This is to ensure a good debugging experience, which requires native files to be excluded from the single file. There is an option to set a flag, `IncludeNativeLibrariesForSelfExtract`, to include native libraries in the single file bundle, but these files will be extracted to a temporary directory in the client machine when the single file application is run. Similarly, single-file doesn't bundle debug files (PDB files) for a better debugging experience.
+
+Managed C++ applications aren't well suited for single-file deployment and we recommend that you write applications in C# or another non-managed C++ language to be single-file compatible. There are some rare cases where a managed library takes a dependency into the runtime library, coreclr, via `pinvoke`. This will not be possible in single-file case since the runtime library is bundled with the application.
+
+The following table outlines the API behavior in a single-file application:
+
+| API                                                                                     | Notes |
```

Comment:
Let's move this up and remove the paragraph explanation for `Assembly.Location`. This info should probably be at the top since everyone will have to review their app for these calls.